tests/cupy_tests/core_tests/test_ndarray_adv_indexing.py

```diff
@@ -191,6 +191,59 @@ def test_adv_getitem_cupy_indices5(self):
         testing.assert_array_equal(original_index, index)
 
 
+@testing.parameterize(
+    {'shape': (9, 16), 'indexes': (
+        numpy.array([8], dtype=numpy.int8),
+        numpy.array([1], dtype=numpy.int8))},
+    {'shape': (9, 3, 16), 'indexes': (
+        numpy.array([8], dtype=numpy.int8),
+        slice(None),
+        numpy.array([1], dtype=numpy.int8))},
+    {'shape': (17, 16), 'indexes': (
+        numpy.array([16], dtype=numpy.uint8),
+        numpy.array([1], dtype=numpy.uint8))},
+    {'shape': (129, 256), 'indexes': (
+        numpy.array([128], dtype=numpy.int16),
+        numpy.array([1], dtype=numpy.int16))},
+    {'shape': (257, 256), 'indexes': (
+        numpy.array([256], dtype=numpy.uint16),
+        numpy.array([1], dtype=numpy.uint16))},
+    {'shape': (129, 256), 'indexes': (
+        numpy.array([128], dtype=numpy.int16),
+        numpy.array([1], dtype=numpy.int32))},
+    {'shape': (129, 256), 'indexes': (
+        numpy.array([128], dtype=numpy.int16),
+        numpy.array([1], dtype=numpy.int8))}
+)
+@testing.gpu
+class TestArrayAdvancedIndexingOverflow(unittest.TestCase):
```

Comment:
Instead of random shapes, please test agains corner cases.
More specifically, we should at least test against some (preferably all) of the following matrix:
* Array size: `a.size` == 2^n-1, 2^n, 2^n+1 where n = 7, 8, 15, 16, 31, 32
* Index dtype:  [u]int{8, 16, 32, 64}
* Index value: 2^n-1, 2^n, 2^n+1 where n = 7, 8, 15, 16, 31, 32

Reply:
Sorry for late reply.
I still have not figured out what is the best way to write tests for this bug.
In my initial PR, I tried to make the test as small as possible.
Here is a intent behind the current code.

---

If we set the size of the array to one of `2^n - 1, 2^n, 2 ^n + 1`, many tests would become redundant unless we set the shape to one dimensional-like, such as `(N, 1)`, because overflow would not occur. You can verify that from the line that causes overflow.
https://github.com/yuyu2172/cupy/blob/8ccbf55da1893f97369e9e7597c72e597af7f16c/cupy/core/core.pyx#L3230

Here is an example with `a.size=2^7, dtype=int8, shape=(2^6, 2)` that demonstrates how overflow does not occur.
The maximum value of `s % a_interm_shape_i` for `i=0` is `2^6 - 1`.
Thus, the maximum value of `stride * (s % a_interm_shape_i)` is `(2^6 - 1) * 2 < 2^7-1`.

The shapes of arrays in my PR are actually not random. I tried to set the size of the array size to be as small as possible while making the overflow to occur.
For instance, with `dtype=int8, shape=(2^3+1, 2^4)`, the maximum value of `stride * (s % a_interm_shape_i)` is `2^4 * 2^3 = 2^7 > 2^7-1` for `i=0`, which is the smallest value to raise overflow.

Reply:
Ah, sorry, I couldn't understand the intent.
Now it's clear that it's not arbitrary values.
Thank you for the fix!